examples/responsive-vis/responsive-vis-example.js

```diff
@@ -0,0 +1,150 @@
+          onClick={() => this.setState({
+            chartType: 'scatterplot',
+            data: createData((~~Math.pow(10, dataSize)), false)
+          })}
+          >
+          Scatterplot
+        </div>
+        <div
+          className={chartType === 'barChart' ? 'selected-chart-type' : 'unselected-chart-type'}
+          onClick={() => this.setState({
+            chartType: 'barChart',
+            data: createData((~~Math.pow(10, dataSize)), true)
+          })}
+          >
+          BarChart
+        </div>
+      </div>
+      {`Data Size: ${(~~Math.pow(10, dataSize))}`}
+      <input
+        onChange={e => {
+          this.setState({
+            dataSize: e.target.value,
+            data: createData(~~Math.pow(10, e.target.value), this.state.chartType === 'barChart')
+          });
+        }}
+        type="range"
+        min={1}
+        max={6}
+        step={0.1}
+        value={dataSize}
+        />
+
+      {`Visualization size: ${visSize}`}
+      <input
+        onChange={e => this.setState({visSize: ~~e.target.value})}
+        type="range"
+        min={100}
+        max={1000}
+        value={visSize}
+        />
+    </div>);
+  }
+
+  render() {
+    const {chartType, data, visSize} = this.state;
+    const margin = {left: 60, top: 60, bottom: 50, right: 50};
+
+    return (
+      <div className="responsive-vis-example">
+        <div className="responsive-explanation">
+          This demo explores the concept of "Responsive Data Visualization"
+          (As coined by Nick Rabinowitz). The basic notion is lifted from responsive design:
+          some features work for some screen resolutions, while others do not.
+          Responsive design determines whether or not to use a given feature by
+          consulting an aspect ratio (width by height). Through this notation we are
+          able to create beautiful web experiences that work seemlessly between phones,
+          tablets, and computers. Taking this idea on step
+          farther we introduce a third element into the fray: data size.
+          <br />
+          <br />
+          In data visualization, we often need to create applications that work with
+          enmorous ranges of sizes of data. Sometimes the data might be small (10 - 100 rows),
+          or it might be gigantic (100k-1M+ row): throughout the entire range it just needs to work.
+          Again, following our cues from responsive design, we note that maybe labels on
+          scatterplots look great when you have under 50 data points, but bad when you have 2000.
+          Checkout Nicks
+          <a href="http://nrabinowitz.github.io/rdv/"> original demo </a>
+           for more details on the theory, as well to see his rad implementation in
+           raw d3.
+
+        </div>
+        <div className="responsive-vis-example-main-content">
+          {this.renderControls()}
+
+          {chartType === 'scatterplot' && <ResponsiveScatterplot
+              data={data}
```

Comment:
Could create an object with common props (all except ref) and pass it to the two components to avoid duplication